prov/sockets/include/sock.h

```diff
@@ -1186,4 +1186,6 @@ int sock_epoll_wait(struct sock_epoll_set *set, int timeout);
 int sock_epoll_get_fd_at_index(struct sock_epoll_set *set, int index);
 void sock_epoll_close(struct sock_epoll_set *set);
 
+size_t sock_rx_avail_len(struct sock_rx_entry *rx_entry);
+
```

Comment:
The function is modified below to change it from an inline.  Given the function, I would move the entire thing into the header file to force inlining.